cpp/src/arrow/engine/substrait/serde.cc

```diff
@@ -86,6 +86,20 @@ Result<std::vector<compute::Declaration>> DeserializePlan(
   return sink_decls;
 }
 
+Result<compute::ExecPlan> DeserializePlan(const Buffer& buf,
+                                          const ConsumerFactory& consumer_factory,
+                                          ExtensionSet* ext_set_out) {
+  ARROW_ASSIGN_OR_RAISE(auto declarations,
+                        DeserializePlans(buf, consumer_factory, ext_set_out));
+  if (declarations.size() > 1) {
+    return Status::Invalid("Substrait plan cannot have multiple root relations");
```

Comment:
```suggestion
    return Status::Invalid("DeserializePlan does not support root relations");
```
Technically a Substrait plan can have multiple root relations.  We just don't support those plans in this method.

Reply:
Yes, right, made the change, thanks!